sas/sas.go

```diff
@@ -201,15 +202,17 @@ func discoverDevices(logger klog.Logger, c *Connector, io ioHandler) error {
 
 	c.Multipath = false
 	c.OSPathName = ""
-	rescaned := false
+	retries := 0
 
 	// two-phase search:
 	// first phase, search existing device paths, if a multipath dm is found, exit loop
 	// otherwise, in second phase, rescan scsi bus and search again, return with any findings
-	for true {
+	for retries < 5 {
 
 		// Find the multipath device using WWN
-		dm, devices = findDiskById(logger, c.VolumeWWN, io)
+		logger.V(5).Info("Sleeping 0.5 seconds before device discovery")
+		time.Sleep(500 * time.Millisecond)
```

Comment:
It'd be good to drop a comment explaining how you came up with this value, e.g. is it based on some concrete detail or is it just something that worked on your system and might need to be larger for a different configuration?

Reply:
Also suggest adding the Sleep time as a const (with a comment)